src/main/java/com/salesforce/dataloader/mapping/LoadMapper.java

```diff
@@ -88,8 +92,8 @@ public void verifyMappingsAreValid() throws MappingInitializationException {
         for (Map.Entry<String, String> entry : getMappingWithUnmappedColumns(false).entrySet()) {
             String sfdcName = entry.getValue();
             if(StringUtils.hasText(sfdcName)) {
-                final Field f = getClient().getField(sfdcName);
-                if (f == null)
+                final HashSet<Field> fieldSet = getClient().getField(sfdcName);
+                if (fieldSet == null | fieldSet.isEmpty())
```

Comment:
What order does the HashSet make?  Why would they need to be in any 
order as long as they're only in the list once?  Using a HashSet to 
communicate there's no order keeps programmers from depending on 
it--which you're right isn't guaranteed.